### PR TITLE
Fix Dropout for `mode = 'test'` case in `jax/example_libraries/stax.py` file

### LINKED ISSUE #16218: Stax has a bug, Dropout cannot convert from training mode to test mode
### Description

Stax has a bug and Dropout cannot convert from training mode to test mode. The correct implementation should be as follows:
```
def Dropout(rate, mode='train'):
  """Layer construction function for a dropout layer with given rate."""
  def init_fun(rng, input_shape):
    return input_shape, ()
  def apply_fun(params, inputs, **kwargs):    
    mode = kwargs.get('mode', None)
    if mode == 'train':
      rng = kwargs.get('rng', None)
      if rng is None:
        msg = ("Dropout layer requires apply_fun to be called with a PRNG key "
              "argument. That is, instead of `apply_fun(params, inputs)`, call "
              "it like `apply_fun(params, inputs, rng)` where `rng` is a "
              "jax.random.PRNGKey value.")
        raise ValueError(msg)
    
      keep = random.bernoulli(rng, rate, inputs.shape)
      return jnp.where(keep, inputs / rate, 0)
    else:
      return inputs
  return init_fun, apply_fun

```

### What jax/jaxlib version are you using?

_No response_

### Which accelerator(s) are you using?

_No response_

### Additional system info

_No response_

### NVIDIA GPU info

_No response_

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
The current `Dropout` implementation unnecessarily checks for the `PRNG key` in `'test' mode`. This pull request resolves this by restricting the `PRNG key` check to `'train' mode` only.

Fixes #16218

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Thanks!
